### ISSUE TITLE
Can one connect to the fingerbot directly using Bluetooth without a hub?

### ISSUE THREAD

**Issue by falhumai96** [jasonacox/tinytuya#350]
I am planning to buy a Fingerbot, but I would like to know if this API allows for a local Bluetooth communication with the device (say, from a raspberry pi).

Thanks.

**Comment by jasonacox** [comment]
Hi @falhumai96 - `tinytuya` uses TCP/IP via WiFi or LAN communication, not bluetooth directly.

**Comment by falhumai96** [comment]
I see, so it seems it requires a Hub to function properly.

**Comment by jasonacox** [comment]
Yes, I suspect that is the case. I have tested using a hub to manage simple bluetooth devices via `tinytuya`.  I don't know about the fingerbot, but would love to know if that works as well.

**Comment by uzlonewolf** [comment]
Although TinyTuya currently does not support it, I don't see why it couldn't be added.  It would take some R&D time to reverse engineer the BLE endpoints, but it should be possible.  Unfortunately I'm out of town and didn't bring a hub or fingerbot with me so I won't be able to look into it until later next month.

**Comment by uzlonewolf** [comment]
Well, I looked into this, and it's not looking too good.  It turns out BT devices use an entirely different protocol and as such I feel it's going to be out of scope for the TinyTuya project.  I was hoping it was going to be the same protocol as the LAN just wrapped in GATT messages, however that is not the case and pretty much none of the existing code can be used in implementing it.  There is already a Fingerbot demo program at https://github.com/redphx/poc-tuya-ble-fingerbot however.

**Comment by jasonacox** [comment]
Bummer. Thanks for exploring this @uzlonewolf !

**Comment by jasonacox** [comment]
Thank you for reporting this issue! This issue has been open for over 2 years with no recent activity. We are closing it to help keep the issue tracker focused and manageable.

If this problem is still present in the latest version of TinyTuya, please feel free to reopen this issue or open a new one with updated details (TinyTuya version, Python version, device type, and steps to reproduce). We'd love to help!

Thank you for your contribution to TinyTuya! 🙏
